### ISSUE TITLE
Fatal error when user CPT for Global Styles is deleted

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#45240]
### Description

If the user CPT for Global Styles is deleted then this line throws a fatal error:

`if ( array_key_exists( 'post_content', $user_cpt ) ) {`

in `wordpress-6.1/class-wp-theme-json-resolver-6-1.php`.

We should check that $user_cpt is an array before we check if a key exists in it.

### Step-by-step reproduction instructions

Delete the user CPT for Global Styles and reload your site.

### Screenshots, screen recording, code snippet

Error:

```
Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /wordpress/plugins/gutenberg/14.3.1/lib/compat/wordpress-6.1/class-wp-theme-json-resolver-6-1.php:84
Stack trace:
#0 /wordpress/plugins/gutenberg/14.3.1/lib/experimental/class-wp-theme-json-resolver-gutenberg.php(203): WP_Theme_JSON_Resolver_6_1::get_user_data()
#1 /wordpress/plugins/gutenberg/14.3.1/lib/experimental/register-webfonts-from-theme-json.php(13): WP_Theme_JSON_Resolver_Gutenberg::get_merged_data()
#2 /wordpress/core/6.0.3/wp-includes/class-wp-hook.php(307): gutenberg_register_webfonts_from_theme_json('')
#3 /wordpress/core/6.0.3/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters(NULL, Array)
#4 /wordpress/core/6.0.3/wp-includes/plugin.php(476): WP_Hook->do_action(Array)
#5 /wordpress/core/6.0.3/wp-settings.php(598): do_action('init')
#6 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1317): require('/wordpress/core...')
#7 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1235): WP_CLI\Runner->load_wordpress()
#8 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
#9 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/bootstrap.php(78): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
#10 phar:///usr/local/bin/wp-cli/vendor/wp-cli/wp-cli/php/wp-cli.php(27): WP_CLI\bootstrap()
#11 phar:///usr/local/bin/wp-cli/php/boot-phar.php(11): include('phar:///usr/loc...')
#12 /usr/local/bin/wp-cli(4): include('phar:///usr/loc...')
#13 {main}
  thrown in /wordpress/plugins/gutenberg/14.3.1/lib/compat/wordpress-6.1/class-wp-theme-json-resolver-6-1.php on line 84
```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by scruffian** [comment]
I have added this to the 6.1 tasks as I wonder if we want to get if fixed in 6.1?

**Comment by ndiego** [comment]
This would be good to fix for 6.1 if a fix can be prepared asap. Otherwise, this is a good candidate for 6.1.1.

**Comment by cbravobernal** [comment]
Taking a look at it 😄

**Comment by ockham** [comment]
TBH I'm having a bit of trouble reproducing. Can someone help me out there?

What I've tried:
- Using TT3:
- Go to the Site Editor.
- Make some Global Style changes, e.g.:
  - Change the Theme style variation to "Sherbet"
  - Change block styling globally, e.g. change Headings to have an orange background color.
 - In the console, list all Global Style CPTs: `npm run wp-env run cli "wp post list --post_type=wp_global_styles"` (Shows only one for me; locate its `ID`)
 - Delete that CPT: `npm run wp-env run cli "wp post delete --force <ID>"`
 - In a browser tab with the frontend open, reload. Note that styles are gone.
 - In the browser tab with the Site Editor open, open the "Network" tab and reload the page.
 - Check REST API requests. _I didn't see any that would report an error_ 🤔 
 - Run `npm run wp-env logs`. _I didn't see any fatals_ 🤔

**Comment by Mamaduka** [comment]
Was anyone able to reproduce this issue with WP 6.1 without the Gutenberg plugin? Because I believe the issue only exists in the plugin.

First, the `get_user_data_from_wp_global_styles` methods in [Core](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json-resolver.php#L402) and [Gutenberg](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.0/class-wp-theme-json-resolver-6-0.php#L165) are different. The method got some optimization during the backport, but we forgot to port changes back to the plugin.

The `get_user_data_from_wp_global_styles` in the plugin can return `null` under certain conditions because it uses a custom caching layer.

**Conditions:**
* The site uses persistent object caching.
* Global Styles post record gets deleted.
* The method returns a cached ID for the deleted Global Styles post.

The cache isn't purged on post deletion but relies on expiration, so the above condition results in `get_post` returning `null`.

**A simple example to fake this condition:**

```php
add_action( 'init', function() {
	$theme = wp_get_theme();

	$args             = array(
		'numberposts' => 1,
		'orderby'     => 'date',
		'order'       => 'desc',
		'post_type'   => 'wp_global_styles',
		'post_status' => array( 'publish' ),
		'tax_query'   => array(
			array(
				'taxonomy' => 'wp_theme',
				'field'    => 'name',
				'terms'    => $theme->get_stylesheet(),
			),
		),
	);

	$cache_key = sprintf( 'wp_global_styles_%s', md5( serialize( $args ) ) );

	// Fake cache with non-existing post ID.
	wp_cache_set( $cache_key, 999999, '', HOUR_IN_SECONDS );

	$data = WP_Theme_JSON_Resolver_6_1::get_user_data_from_wp_global_styles( $theme );

	// The $data will be null.
	var_dump( $data );
} );
```

Cc @hellofromtonya, @desrosj

**Comment by Mamaduka** [comment]
This was fixed in the plugin via #45634.
